Source/Testably.Abstractions.Testing/FileSystem/FileSystemWatcherMock.cs

```diff
@@ -568,10 +562,39 @@ private void TriggerRenameNotification(ChangeDescription item)
 			}
 			else if (item.OldPath != null)
 			{
-				Deleted?.Invoke(this, ToFileSystemEventArgs(
-					WatcherChangeTypes.Deleted, item.OldPath, item.OldName));
-				Created?.Invoke(this, ToFileSystemEventArgs(
-					WatcherChangeTypes.Created, item.Path, item.Name));
+				string fullPath = _fileSystem.Execute.Path.GetFullPath(Path);
+				if (IncludeSubdirectories)
+				{
+					if (item.OldPath.StartsWith(fullPath, _fileSystem.Execute.StringComparisonMode))
+					{
+						Deleted?.Invoke(this, ToFileSystemEventArgs(
+							WatcherChangeTypes.Deleted, item.OldPath, item.OldName));
+					}
+					if (item.Path.StartsWith(fullPath, _fileSystem.Execute.StringComparisonMode))
+					{
+						Created?.Invoke(this, ToFileSystemEventArgs(
+							WatcherChangeTypes.Created, item.Path, item.Name));
+					}
```

Comment:
Deleted gets invoked if `OldPath` matches the watcher's path. Created gets invoked if `Path` matches the watcher's path.